awscli/clidriver.py

```diff
@@ -379,15 +366,17 @@ def __call__(self, args, parsed_globals):
 
     def _create_command_table(self):
         command_table = OrderedDict()
-        service_object = self._get_service_object()
-        for operation_object in service_object.operations:
-            cli_name = xform_name(operation_object.name, '-')
+        service_model = self._get_service_model()
```

Comment:
I'm not sure I'm really a fan of adding a `service_model` property to this class. I'd like to move in a general direction towards unifying the custom commands and the clidriver command code paths.  In order to do this, all the command objects should have a consistent interface.  For many custom commands the concept of "service model" doesn't really apply.

From what I can tell, this is only used in waiters.  Maybe there's a way to implement waiters without having to expose service_model in the ServiceCommand objects.